src/Tribe/REST/V1/Endpoints/Ticket_Archive.php

```diff
@@ -29,36 +29,56 @@ public function get_documentation() {
 	 * @return WP_Error|WP_REST_Response An array containing the data on success or a WP_Error instance on failure.
 	 */
 	public function get( WP_REST_Request $request ) {
-		$data       = array();
 		$query_args = $request->get_query_params();
-		$per_page   = $request->get_param( 'per_page' );
-		$found      = array();
+		$per_page   = (int) $request->get_param( 'per_page' );
+		$page       = (int) $request->get_param( 'page' );
 
-		$this->ticket_query_args['paged']          = $request->get_param( 'page' );
-		$this->ticket_query_args['posts_per_page'] = $request->get_param( 'per_page' );
-		$this->found_tickets                       = 0;
+		$fetch_args = array();
 
 		if ( $request->get_param( 'include_post' ) ) {
-			$include_post = $request['include_post'];
+			$include_post               = $request['include_post'];
+			$fetch_args['event']        = $include_post; // by( 'event' ,$id )
+			$query_args['include_post'] = implode( ',', $include_post );
+		}
 
-			foreach ( $include_post as $post_id ) {
-				$found[] = $this->get_tickets_for_post( $post_id );
-			}
+		/** @var wpdb $wpdb */
+		global $wpdb;
 
-			$found = call_user_func_array( 'array_merge', $found );
+		if ( current_user_can( 'read_private_posts' ) ) {
+			$permission                = Tribe__Tickets__REST__V1__Read_Repository::PERMISSION_EDITABLE;
+			$fetch_args['post_status'] = 'any';
+		} else {
+			$permission                = Tribe__Tickets__REST__V1__Read_Repository::PERMISSION_READABLE;
+			$fetch_args['post_status'] = 'publish';
+		}
 
-			$query_args['include_post'] = implode( ',', $include_post );
+		$found = tribe_tickets( 'restv1' )
+			->fetch()
+			->by_args( $fetch_args )
+			->permission( $permission )
+			->found();
```

Comment:
If [this comment](https://github.com/moderntribe/tribe-common/pull/730/files#r199812738) has merit, I'd say `count()` would be better than `found()` in this case (and subsequently `$count` would be preferred over `$found`).